src/main/java/org/springframework/data/couchbase/core/RxJavaCouchbaseTemplate.java

```diff
@@ -102,6 +113,74 @@
                 .flatMap(object -> remove(object, persistTo, replicateTo));
     }
 
+    @Override
+    public <T> Observable<T> cas(String id, Class<T> entityClass, Function<Optional<T>, CasResult<T>> updateFunction) {
+        return cas(id, entityClass, PersistTo.NONE, ReplicateTo.NONE, updateFunction);
+    }
+
+    @Override
+    public <T> Observable<T> cas(String id, Class<T> entityClass, PersistTo persistTo, ReplicateTo replicateTo, Function<Optional<T>, CasResult<T>> updateFunction) {
+        return asyncCas(id, entityClass, persistTo, replicateTo, old -> Observable.just(updateFunction.apply(old)));
+    }
+
+    @Override
+    public <T> Observable<T> asyncCas(String id, Class<T> entityClass, Function<Optional<T>, Observable<CasResult<T>>> updateFunction) {
+        return asyncCas(id, entityClass, PersistTo.NONE, ReplicateTo.NONE, updateFunction);
+    }
+
+    @Override
+    public <T> Observable<T> asyncCas(String id, Class<T> entityClass, PersistTo persistTo, ReplicateTo replicateTo, Function<Optional<T>, Observable<CasResult<T>>> updateFunction) {
+        return client.get(id, RawJsonDocument.class)
+                .flatMap(doc -> update(doc, entityClass, persistTo, replicateTo, updateFunction))
+                .switchIfEmpty(insert(entityClass, persistTo, replicateTo, updateFunction))
+                .retryWhen(casRetryFunction())
+                .onErrorResumeNext(e -> Observable.error(e.getCause()))
+                .doOnError(e -> TemplateUtils.translateError(e));
```

Comment:
I don't think we need to do this in the library. The cas retries can be chained in by the application itself.